crates/subspace-farmer/src/farmer_cache.rs

```diff
@@ -342,13 +290,8 @@ where
         info!("Initializing piece cache");
 
         // Pull old cache state since it will be replaced with a new one and reuse its allocations
-        let PieceCachesState {
-            mut stored_pieces,
-            mut dangling_free_offsets,
-            backends: _,
-        } = mem::take(&mut *self.piece_caches.write().await);
-        stored_pieces.clear();
-        dangling_free_offsets.clear();
+        let (mut stored_pieces, mut dangling_free_offsets) =
+            mem::take(&mut *self.piece_caches.write().await).clear();
```

Comment:
The previous code cleared stored_pieces, but this code clears the empty cache. (take() replaces the value with default() - the empty cache, and then returns the original value.)

Is this change intentional?

Reply:
There is no actual modification here, clear is executed on the Caches that are taken out, the naming of clear is indeed a bit ambiguous, so let me fix it.

Reply:
I see, yes, you're right!